src/main/java/de/tum/in/www1/artemis/web/rest/LearningGoalResource.java

```diff
@@ -197,31 +197,30 @@ public ResponseEntity<LearningGoal> updateLearningGoal(@PathVariable Long course
         if (learningGoal.getId() == null) {
             throw new BadRequestException();
         }
-        var learningGoalFromDb = this.learningGoalRepository.findByIdWithLectureUnitsBidirectionalElseThrow(learningGoal.getId());
-        if (learningGoalFromDb.getCourse() == null || !learningGoalFromDb.getCourse().getId().equals(courseId)) {
+        var existingLearningGoal = this.learningGoalRepository.findById(learningGoal.getId()).orElseThrow();
```

Comment:
Is there a reason you are not using `learningGoalRepository.findByIdElseThrow` here?

Reply:
Fixed.